assets/js/veracode.js

```diff
@@ -1,100 +1,155 @@
-jQuery(function () {
-  const rootDataUrl = '/reports/veracode/data'
-  const columns = [
-    {
-      data: 'name',
-      createdCell: function (td, _cellData, rowData) {
-        $(td).html(`<a href="/components/${cleanColumnOutput(rowData.name)}">${cleanColumnOutput(rowData.name)}</a>`)
+class VeracodeRenderer {
+  constructor(csrfToken, viewMode) {
+    this.csrfToken = csrfToken
+    this.viewMode = viewMode
+  }
+
+  post = async (url, body) => {
+    const response = await fetch(url, {
+      method: 'POST',
+      credentials: 'same-origin',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+        'X-CSRF-Token': this.csrfToken,
       },
+      body: JSON.stringify(body),
+    })
+    if (!response.ok) {
+      throw new Error(`There was a problem calling: ${url}`)
+    }
+    return response.json()
+  }
+
+  fetchMessages = async componentNames => {
+    const vulnerabilities = await this.post('/reports/veracode/data', { componentNames })
```

Comment:
So moving this into `/veracode` route means updating this URL as well as moving the code otherwise it's pointing to the old route.